### ISSUE TITLE
Can we calculate student and teacher outputs just once for training the generator?

### ISSUE THREAD

**Issue by vinsis** [polo5/ZeroShotKnowledgeTransfer#4]
Hi, I have a question regarding the implementation in [these lines](https://github.com/polo5/ZeroShotKnowledgeTransfer/blob/master/solver.py#L94-L101).

Since you are training the generator in these lines, `x_pseudo`, `self.teacher` and `self.student` do not change. This means that `student_logits`, `student_activations`, `teacher_logits` and `teacher_activations` do not change. Thus they only need to be calculated once. Can you please confirm if this is truly the case or if I misunderstood something? Thank you.

**Comment by vinsis** [comment]
Ah, it is in order to calculate the gradients. Closing it now.
